app/decorators/physical_switch_decorator.rb

```diff
@@ -0,0 +1,42 @@
+class PhysicalSwitchDecorator < MiqDecorator
+  def self.fonticon
+    'ff ff-network-switch'
+  end
+
+  def quadicon
+    {
+      :top_left     => {
+        :text    => (hardware.physical_ports ? hardware.physical_ports.size : 0),
+        :tooltip => 'Ports'
+      },
+      :top_right    => {
+        :tooltip => power_state.try(:downcase),
+      }.merge(QuadiconHelper.machine_state(power_state)),
+      :bottom_left  => {
+        :fonticon => fonticon,
+        :tooltip  => type,
+      },
+      :bottom_right => {
+        :fileicon => health_state_img,
+        :tooltip  => health_state,
+      }
+    }
+  end
+
+  def single_quad
+    {
+      :fonticon => fonticon
+    }
+  end
+
+  private
+
+  def health_state_img
+    case health_state
+    when "Valid"    then "svg/healthstate-normal.svg"
+    when "Critical" then "svg/healthstate-critical.svg"
+    when "Warning"  then "100/warning.png"
+    else "svg/healthstate-unknown.svg"
+    end
+  end
```

Comment:
These are okay for now, just note for myself that I have to take a look at this location when converting these to the fonticon+color+background combo in the future.